src/platform/plugins/shared/metrics_experience/server/routes/dimensions/get_dimensions.ts

```diff
@@ -28,16 +38,25 @@ export const getDimensions = async ({
   from,
   to,
   logger,
+  query: originalQuery,
 }: CreateDimensionsParams): Promise<Array<{ value: string; field: string }>> => {
   if (!dimensions || dimensions.length === 0) {
     return [];
   }
 
+  const whereCommand = originalQuery
+    ? Walker.find(
+        Parser.parse(originalQuery).root,
```

Comment:
`Parser.parse` could throw an exception in case the query is malformed. Should we handle this?